lib/sles4sap/sap_deployment_automation_framework/configure_workload_tfvars.pm

```diff
@@ -0,0 +1,395 @@
+'WORKLOAD-ZONE' tfvars file.
+Generated file is following example template:
+https://github.com/Azure/SAP-automation-samples/blob/main/Terraform/WORKSPACES/SYSTEM/LAB-SECE-SAP04-L00/LAB-SECE-SAP04-L00.tfvars
+
+=cut
+
+our @EXPORT = qw(
+  create_workload_tfvars
+  write_tfvars_file
+);
+
+=head2 write_tfvars_file
+
+    write_tfvars_file(tfvars_file=>'/path/towards/happiness', tfvars_data=>$tfvars_data);
+
+Data provided in B<$args{tfvars_data}> is compiled into tfvars file format and written into target B<$args{tfvars_file}>
+file located on deployer VM.
+B<$args{tfvars_data}> is a HASHREF containing individual sections that should be included in final tfvars content.
+B<file_header> is placed as a comment as a first line of the defined section.
+
+B<Example:>
+{
+file_header => "Comment placed on top of the tfvars file",
+env_definitions => {header => 'Comment placed on top of a section',
+    tfvars_variable = 'value', tfvars_variable_2 = 'value2'},
+workload_networking => {header => 'Comment placed on top of a section',
+    tfvars_variable_3 = 'value3', tfvars_variable_4 = 'value4'},
+}
+
+=over
+
+=item * B<tfvars_file>: Target tfvars file location
+
+=item * B<tfvars_data>: Target tfvars file data - Must be a HASHREF
+
+=item * B<section_order>: Order in which the sections should be compiled into the file - must be an ARRAYREF
+
+=back
+
+=cut
+
+sub write_tfvars_file {
+    my (%args) = @_;
+    for my $arg ('tfvars_data', 'tfvars_file') {
+        croak "Missing mandatory argument \$args{$arg}" unless $args{$arg};
+    }
+    croak 'Argument \$args{tfvars_data} must be a HASHREF' unless ref($args{tfvars_data}) eq 'HASH';
+    if ($args{section_order}) {
+        croak 'Argument \$args{section_order} must be a ARRAYREF' unless ref($args{section_order}) eq 'ARRAY';
+    }
+
+    # If $args{section_order} is not defined, just use list of keys.
+    my @section_order = $args{section_order} ? @{$args{section_order}} : keys(%{$args{tfvars_data}});
+    my $file_contents = $args{tfvars_data}->{file_header};
+    for my $section (@section_order) {
+        next if $section eq 'file_header';
+        $file_contents .= compile_tfvars_section($args{tfvars_data}->{$section});
+    }
+
+    write_sut_file($args{tfvars_file}, $file_contents);
+}
+
+=head2 compile_tfvars_section
+
+    compile_tfvars_section($section_data);
+
+Converts HASHREF based tfvars section into terraform format and returns it as a string.
+Results in:
+# Header
+variable_1 = 'value_1'
+variable_2 = 'value_2'
+
+=over
+
+=item * B<$section_data>: HASHREF containing tfvars section header (comment) and variables
+
+=back
+
+=cut
+
+sub compile_tfvars_section {
+    my ($section_data) = @_;
+    croak 'Argument \$section_data must be a HASHREF' unless ref($section_data) eq 'HASH';
+
+    my $header = $section_data->{header};
+    # Remove header from hash otherwise it will be turned into blank line by map
+    delete $section_data->{header};
```

Comment:
Suggestion:
```
    # Remove header from hash otherwise it will be turned into blank line by map
    my $header = delete $section_data->{header};
```

Reply:
Oh I missed this info in docs. 
```
In list context, usually returns the value or values deleted
```
Thanks. I'll do it that way.